lib/Components/AsyncComponent.js

```diff
@@ -0,0 +1,65 @@
+import {AbstractComponent} from './AbstractComponent';
+
+export class AsyncComponent extends AbstractComponent {
+
+    /**
+     * @param {Element} [element]
+     */
+    render(element) {
+        if (!this.isRendered()) {
+            let componentPromise = this.options.promise,
+                renderer = this.getRenderer(),
+                env = this.getEnv();
+
+            this.element = element || renderer.createComment();
+
+            if (componentPromise) {
+                let promise = componentPromise.then((component) => {
+                    if (!this.isRendered()) {
+                        return;
+                    }
+
+                    this.component = component;
+
+                    // if component is comment, avoid restore
+                    if (element && element.nodeType === Node.COMMENT_NODE) {
```

Comment:
Please take into account that Node is defined only in browser. If this line possibly can be executed on server then Node should be mocked with enum